File: Server/src/main/content/global/skill/production/cooking/recipe/potato/impl/EggPotato.java
package content.global.skill.production.cooking.recipe.potato.impl;

import content.global.skill.production.cooking.recipe.potato.PotatoRecipe;
import core.game.node.item.Item;

/**
 * The Egg potato.
 */
public class EggPotato extends PotatoRecipe {

    private static final Item EGG_POTATO = new Item(7056);

    private static final Item TOPPING = new Item(7064);

    @Override
    public Item getProduct() {
        return EGG_POTATO;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{TOPPING};
    }

    @Override
    public boolean isTopping() {
        return true;
    }

    @Override
    public int getLevel() {
        return 51;
    }

    @Override
    public double getExperience() {
        return 10;
    }

}

File: Server/src/main/content/global/skill/production/cooking/recipe/potato/impl/EggPotato.kt
package content.global.skill.production.cooking.recipe.potato.impl

import content.global.skill.production.cooking.recipe.potato.PotatoRecipe
import core.api.consts.Items
import core.game.node.item.Item

class EggPotato : PotatoRecipe() {

    override fun getProduct(): Item {
        return EGG_POTATO
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(TOPPING)
    }

    override fun isTopping(): Boolean {
        return true
    }

    override fun getLevel(): Int {
        return 51
    }

    override fun getExperience(): Double {
        return 10.0
    }

    companion object {
        private val EGG_POTATO = Item(Items.EGG_POTATO_7056)
        private val TOPPING = Item(Items.EGG_AND_TOMATO_7064)
    }
}
